src/AutoRest.CSharp/Common/Generation/Writers/DocumentationWriterExtensions.cs

```diff
@@ -68,6 +70,48 @@ public static CodeWriter WriteXmlDocumentationReturns(this CodeWriter writer, Fo
             return writer.WriteDocumentationLines($"<returns>", $"</returns>", text);
         }
 
+        public static CodeWriter WriteXmlDocumentationInclude(this CodeWriter writer, string filename, MethodSignature methodSignature, out string memberId)
+        {
+            // We use short names of types for external doc reference member id
+            // This is not good enough for cref, but for now works as member id
+            // Change to cref-style names if needed
+            var sb = new StringBuilder();
+            sb.Append(methodSignature.Name).Append("(");
+            foreach (var parameter in methodSignature.Parameters)
+            {
+                AppendTypeWithShortNames(parameter.Type, sb);
+                sb.Append(",");
+            }
+
+            sb.Remove(sb.Length - 1, 1);
+            sb.Append(")");
+
+            memberId = sb.ToString();
+            return writer.LineRaw($"/// <include file=\"{filename}\" path=\"doc/members/member[@name='{memberId}']/*\" />");
```

Comment:
does this count as "writing documentations"? should we have a check if the writer is in the mode of writing documents?

Reply:
No, we lie here to the code writer on purpose. `include` doesn't follow our type formatting rules, hence a separate method that doesn't call `AppendXmlDocumentation` and the string is fully constructed and added via `LineRaw`.